src/DataBox/DataBox/Common/Cmdlets/GetDataBoxJobCmdlet.cs

```diff
@@ -0,0 +1,176 @@
+                result.Add(new PSDataBoxJob(JobsOperationsExtensions.Get(
+                                this.DataBoxManagementClient.Jobs,
+                                this.ResourceGroupName,
+                                this.Name,
+                                "details")));
+                WriteObject(result,true);
+               
+            }
+            else if (!string.IsNullOrEmpty(this.ResourceGroupName))
+            {
+                IPage<JobResource> jobPageList = null;
+                List<JobResource> result = new List<JobResource>();
+                List<PSDataBoxJob> finalResult = new List<PSDataBoxJob>();
+                
+                do
+                {
+                    // Lists all the jobs available under resource group.
+                    if (jobPageList == null)
+                    {
+                        jobPageList = JobsOperationsExtensions.ListByResourceGroup(
+                                        this.DataBoxManagementClient.Jobs,
+                                        this.ResourceGroupName);
+                    }
+                    else
+                    {
+                        jobPageList = JobsOperationsExtensions.ListByResourceGroupNext(
+                                        this.DataBoxManagementClient.Jobs,
+                                        jobPageList.NextPageLink);
+                    }
+
+                    
+                    if (Completed || Cancelled || Aborted || CompletedWithErrors)
+                    {
+                        foreach (var job in jobPageList)
+                        {
+                            if ((Completed && job.Status == StageName.Completed) 
+                                || (Cancelled && job.Status == StageName.Cancelled) 
+                                || (Aborted && job.Status == StageName.Aborted)
+                                || (CompletedWithErrors && job.Status == StageName.CompletedWithErrors))
+                            {
+                                result.Add(job);
+                            }
+                        }
+                    }
+                    else
+                    {
+                        result.AddRange(jobPageList.ToList());
+                    }
+
+                } while (!(string.IsNullOrEmpty(jobPageList.NextPageLink)));
+                foreach(var job in result)
+                {
+                    finalResult.Add(new PSDataBoxJob(job));
+                }
+                WriteObject(finalResult, true);
+            }
+            else
+            {
+
+                 IPage<JobResource> jobPageList = null;
+                 List<JobResource> result = new List<JobResource>();
+                 List<PSDataBoxJob> finalResult = new List<PSDataBoxJob>();
+
+
+                do
+                {
+                     // Lists all the jobs available under the subscription.
+                     if (jobPageList == null)
```

Comment:
seems like you can use a single method to expland pages and filter the list, if you just pass in a function to get the job page list

Reply:
I did not understand. Anyways is this not okay to do?